dao/src/main/java/se/fortnox/reactivewizard/db/query/parts/ParamQueryPart.java

```diff
@@ -0,0 +1,134 @@
+package se.fortnox.reactivewizard.db.query.parts;
+
+import se.fortnox.reactivewizard.dates.Dates;
+import se.fortnox.reactivewizard.db.query.PreparedStatementParameters;
+import se.fortnox.reactivewizard.json.JsonSerializerFactory;
+import se.fortnox.reactivewizard.util.ReflectionUtil;
+import se.fortnox.reactivewizard.util.rx.PropertyResolver;
+
+import java.lang.reflect.Type;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class ParamQueryPart implements DynamicQueryPart {
+    private static final JsonSerializerFactory JSON_SERIALIZER_FACTORY = new JsonSerializerFactory();
+
+    protected final PropertyResolver             argResolver;
+    protected final int                          argIndex;
+    private final   PreparedStatementParamSetter paramSetter;
+
+    public ParamQueryPart(int argIndex, Type cls) throws SQLException {
+        this(argIndex, ReflectionUtil.getPropertyResolver(cls, new String[0]).get());
+    }
+
+    protected ParamQueryPart(ParamQueryPart paramQueryPart) throws SQLException {
+        this(paramQueryPart.argIndex, paramQueryPart.argResolver);
+    }
+
+    protected ParamQueryPart(int argIndex, PropertyResolver argResolver) throws SQLException {
+        this.argIndex = argIndex;
+        this.argResolver = argResolver;
+        paramSetter = createParamSetter(argResolver.getPropertyGenericType());
+    }
+
+    @Override
+    public void visit(StringBuilder sql, Object[] args) {
+        sql.append("?");
+    }
+
+    @Override
+    public void addParams(PreparedStatementParameters parameters, Object[] args) throws SQLException {
+        Object val = getValue(args);
+        if (val == null) {
+            parameters.addNull();
+        } else {
+            paramSetter.call(parameters, val);
+        }
+    }
+
+    protected Object getValue(Object[] args) {
+        return argResolver.getValue(args[argIndex]);
+    }
+
+    @Override
+    public DynamicQueryPart subPath(String[] subPath) throws SQLException {
+        Optional<PropertyResolver> propertyResolver = argResolver.subPath(subPath);
+        if (!propertyResolver.isPresent()) {
+            throw new RuntimeException("Properties " + subPath + " cannot be found in " + argResolver);
+        }
+        return new ParamQueryPart(argIndex, propertyResolver.get());
+    }
+
+    protected PreparedStatementParamSetter createParamSetter(Type type) throws SQLException {
+        Class<?> rawType = ReflectionUtil.getRawType(type);
+        if (java.util.Date.class.isAssignableFrom(rawType)) {
```

Comment:
This is the only usage of java.util.Date, which required the dates module to be migrated as well. I vote for removing this support now, and dropping the dates module in this PR.